### PR TITLE
Latitude and Longitude Functions

### PR DESCRIPTION
**Pull request by situx**
As suggested in #160 I am proposing the inclusion of a latitude and a longitude function for GeoSPARQL.

We may have two possibilities for definition here:
1. Define the function for points only (which was also implemented in the Apache Jena extension)
2. Define the function for all geometries. For geometries other than points find a solution as to which latitude/longitude to return. My current proposal says "return the latitude/longitude of the centroid of the given geometry" but that is just one possible proposal.

@nicholascar and @SimonBin any opinions?

### CODE REVIEW

**Comment by SimonBin** [comment]
I think doing 1. is fine, as for prior art we have e.g. MySQL: https://dev.mysql.com/doc/refman/8.4/en/gis-point-property-functions.html#function_st-latitude 

if the user desires the outcome of 2., they can call geof:centroid manually

File: spec/sections/11-geometry_extension.adoc [hunk 1]
```diff
@@ -1632,6 +1634,23 @@ The function http://www.opengis.net/def/function/geosparql/length[`geof:length`]
 See the <<Recommendation for units of measure, Recommendation for units of measure>>.
 ====
 
+[%unnumbered]
+[source,turtle]
+----
+geof:latitude (geom: ogc:geomLiteral): xsd:double
+----
+
+The function http://www.opengis.net/def/function/geosparql/latitude[`geof:longitude`] returns the latitude of `geom` in meters. If the geometry is a point it returns the point's latitude. If the geometry is a more complex geometry it returns the latitude of its centroid as calculated by geof:centroid.
```
Comment (SimonBin) [inline spec/sections/11-geometry_extension.adoc:1643]: in metrers -> in degree?

Comment (situx) [reply spec/sections/11-geometry_extension.adoc:1643]: How about "returns the latitude of `geom` in the unit that is proposed by the coordinate reference system associated with the geometry"?

Comment (SimonBin) [reply spec/sections/11-geometry_extension.adoc:1643]: from my feeble understanding, lat and lon would only ever be defined as an angle (normal to the surface and plane of the celestial body resp. rotational angle between the zero meridian), and thus only degree (or radian??) would be suitable. The other question is related to #398 i.e. should a point defined on a metric (projected) CRS be eligible for automatic transformation into CRS:84 in order to determine its lat/lon on the celestial body or not

**Comment by jabhay** [comment]
Discussion in the group:

Should this be for points only? Options presented were:

1. Define the function for points only (which was also implemented in the Apache Jena extension)
2. Define the function for all geometries. For geometries other than points find a solution as to which latitude/longitude to return. My current proposal says "return the latitude/longitude of the centroid of the given geometry" but that is just one possible proposal.

Option 1 was selected as there is already a function provided to get the centroid of a shape other than a point. We don't want to make assumptions over how to chain this together to produce a result for non-Points given the functions required for the user to decide that are available.

How should the centroid be returned? Options presented:

1. Always a particular unit.
2. In the unit of the CRS.

Option 2 was selected as it is aligned with how Simple Features functions work.

**Comment by nicholascar** [comment]
1. Please add the function definition to the [B.1.  Functions Summary Table](https://docs.ogc.org/is/22-047r1/22-047r1.html#_8646c885-ac65-42bf-bfa9-8d170fa7b4e9)
2. Please provide an example in [Annex C - GeoSPARQL Examples](https://docs.ogc.org/is/22-047r1/22-047r1.html#_3e570415-5147-4114-a8f9-6dcda569c0fd)

**Comment by situx** [comment]
Meeting 26th June: We decided to rename the functions to geof:easting and geof:northing so that values in meter may be returned

**Comment by situx** [comment]
- Updated this pull request to use geof:easting and geof:northing
- Added an example in Annex C
- Added the functions to the Functions table
- Added a file that reprents the RDF used in Annex C to the examples folder

**Review by nicholascar — APPROVED** [review]
Just curious why the example data is in RDF/XML as opposed to Turtle??

**Comment by situx** [comment]
I agree, we could add a TTL version. It is RDF/XML because in the GeoSPARQL 1.0 spec in the Annex the example is also RDF/XML. In the GeoSPARQL Compliance Benchmark we also chose to stick to RDF/XML for consistency

**Review by jabhay — APPROVED** [review]
Looks like what we discussed has been implemented.

**Comment by situx** [comment]
I added the TTL version of the example from Annex C. With that I will now merge this pull request.

**Pull request merged**